### ISSUE TITLE
PolyChord (hypercube samplers) with joint priors

### ISSUE THREAD

**Issue by gvcallen** [gvcallen/paramrf#195]
## What to build

Hypercube samplers (PolyChord) with joint priors, following ADR-0005 (`docs/adr/0005-priors-by-name.md`). With this ticket, a trained normalizing flow over fit 1's parameters can be the PolyChord prior of fit 2.

- The hypercube branch of `prf.infer.sample` builds the prior transform for each joint-prior block as `u -> Normal(0, 1).icdf(u) -> whitening -> raw -> declared`. This applies whenever the whitening ends in an independent standard-normal base.
- Parameters not under a joint prior keep their own `icdf`.
- Every cube point maps inside the parameters' bounds. With `space='raw'` this holds through the old raw-to-declared mappings. A declared- or physical-space joint prior is only accepted by `prf.prior` when its support lies inside the bounds (#194). So the transform needs no clipping or rejection, and the evidence is exact.
- A joint prior without such a base raises. The error message says which structure is needed. It replaces today's message, which says to use a joint or split sampler for a joint prior.
- The map is built in ParamRF's sampler code, not as a distribution `icdf`.
- Starting cube values (`u0`, `init_samples`) are mapped the other way consistently.
- A one-dimensional prior attached with `prf.prior(..., space='raw')`, or `space='physical'` on a scaled parameter, is a `Transformed` distribution with no `icdf`. Today a hypercube sampler raises a bare `NotImplementedError` for it. Map it through its base's `icdf` and its bijector when the base has one; otherwise raise a clear error naming the parameter and the space.

Seam: `pmrf/infer/base.py`.

## Acceptance criteria

- [ ] For the correlated-Gaussian example below, uniform cube draws pushed through the prior transform reproduce `mu` and `L Lᵀ` in raw space, within the stated tolerances. `c.C` follows its own prior.
- [ ] The same holds for a small `fleqx.coupling_flow`, checked against samples drawn from the flow directly. This only applies if fleqx is available in the test environment; otherwise the test is skipped with a reason.
- [ ] A joint prior whose whitening has no standard-normal base raises the new message.
- [ ] Mapping a starting model to the cube and back returns it.
- [ ] Cube draws for the correlated-Gaussian example all land inside every parameter's bounds.
- [ ] A one-dimensional `prf.prior` in `'raw'` or `'physical'` space samples correctly under a hypercube sampler when its base has an `icdf`, and otherwise raises a clear error naming the parameter.

## The correlated-Gaussian example

An affine bijector stands in for a trained flow. It has the same structure (a bijector over a standard-normal base) and a closed-form answer. From the handoff that motivated ADR-0005:

```python
model = {
    'a': Resistor(R=prf.Random(RTNormal(50.0, 0.1)), name='a'),
    'b': Resistor(R=prf.Random(RTNormal(50.0, 0.1)), name='b'),
    'c': Capacitor(C=prf.Random(RTNormal(1.0, 0.1), scale=1e-12), name='c'),
}
names = ('a.R', 'b.R')                       # c.C keeps its own prior
mu = jnp.array([3.9, 3.9])
L = jnp.array([[0.10, 0.0], [0.08, 0.06]])   # correlation 0.8
dist = Transformed(Independent(Normal(jnp.zeros(2), jnp.ones(2))), Chain([Block(Shift(mu), 1), TriangularLinear(L)]))
model = prf.prior(model, names, dist, space='raw')
```

## Blocked by

- #194
